app/controllers/development_metrics_controller.rb

```diff
@@ -167,12 +168,23 @@ def code_climate_repository_summary
   def code_climate_department_summary
     CodeClimate::RepositoriesSummaryService.call(
       department: department,
-      from: metric_params[:period],
+      from: metric_params[:from],
+      to: metric_params[:to],
       technologies: []
     )
   end
 
   def department_overview
-    Builders::DepartmentOverview.call(department, from: metric_params[:period].to_i.weeks.ago)
+    Builders::DepartmentOverview.call(department, from: @from, to: @to)
+  end
+
+  def validate_dates
+    @from = metric_params[:from]
```

Comment:
minor comment, but I don't like having any kind of logic on the controllers (for maintainability). So maybe you can have a `DateValidator`, `DateValidatorService`, or something like that that performs the validations. That approach would also help with duplication. I know it might seem an overkill for just 4 lines, so yeah, up to you